test/extended/default_forcePull_test.go

```diff
@@ -0,0 +1,262 @@
+	Expect(cerr).NotTo(HaveOccurred())
+}
+
+func resetImage() {
+	// reset both images just to be sure
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Calling docker tag to reset images", time.Now().Format(time.RFC850)))
+
+	serr := exutil.TagImage(s2iDockBldr, s2iStartingHexID)
+
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Reset for %s to %s complete with err %v", time.Now().Format(time.RFC850), s2iStartingHexID, s2iDockBldr, serr))
+	Expect(serr).NotTo(HaveOccurred())
+
+	cerr := exutil.TagImage(custBldr, custStartingHexID)
+
+	By(fmt.Sprintf("\n%s****FORCE PULL*****:  Reset for %s to %s complete with err %v", time.Now().Format(time.RFC850), custStartingHexID, custBldr, cerr))
+	Expect(cerr).NotTo(HaveOccurred())
+
+}
+
+func verifyImagesSame(strategy string) {
+	s2iComp := s2iDockBldr + ":latest"
+	custComp := custBldr + ":latest"
+
+	comps := []string{s2iComp, custComp}
+	retIDs, gerr := exutil.GetImageIDForTags(comps)
+
+	Expect(gerr).NotTo(HaveOccurred())
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s false ******:  compare image - %s, %s, %s, %s", time.Now().Format(time.RFC850), strategy, s2iComp, custComp, retIDs[0], retIDs[1]))
+	Expect(len(retIDs[0]) > 0).Should(BeTrue())
+	Expect(len(retIDs[1]) > 0).Should(BeTrue())
+	Expect(retIDs[0]).To(Equal(retIDs[1]))
+}
+
+func verifyImagesDifferent(strategy string) {
+	s2iComp := s2iDockBldr + ":latest"
+	custComp := custBldr + ":latest"
+
+	comps := []string{s2iComp, custComp}
+	retIDs, gerr := exutil.GetImageIDForTags(comps)
+
+	Expect(gerr).NotTo(HaveOccurred())
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s true ******:  compare image - %s, %s, %s, %s", time.Now().Format(time.RFC850), strategy, s2iComp, custComp, retIDs[0], retIDs[1]))
+	Expect(len(retIDs[0]) > 0).Should(BeTrue())
+	Expect(len(retIDs[1]) > 0).Should(BeTrue())
+	Expect(retIDs[0] != retIDs[1]).Should(BeTrue())
+}
+
+func waitForBuild(context string, oc *exutil.CLI) {
+	buildName := "ruby-sample-build-1"
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s ******:   waiting for %s", time.Now().Format(time.RFC850), context, buildName))
+	exutil.WaitForABuild(oc.REST().Builds(oc.Namespace()), buildName,
+		// The build passed
+		func(b *buildapi.Build) bool {
+			return b.Name == buildName && b.Status.Phase == buildapi.BuildPhaseComplete
+		},
+		// The build failed
+		func(b *buildapi.Build) bool {
+			if b.Name != buildName {
+				return false
+			}
+			return b.Status.Phase == buildapi.BuildPhaseFailed || b.Status.Phase == buildapi.BuildPhaseError
+		},
+	)
+	// do not care if build returned an error ... entirely possible ... we only check if the image was updated or left the same appropriately
+	By(fmt.Sprintf("\n%s******FORCE PULL TEST %s  ******:   done waiting for %s", time.Now().Format(time.RFC850), context, buildName))
+}
+
+func doBuild(json string, oc *exutil.CLI) {
+	err := oc.Run("create").Args("-f", json).Execute()
+	Expect(err).NotTo(HaveOccurred())
+	_, berr := oc.Run("start-build").Args("ruby-sample-build").Output()
```

Comment:
Yeah, when there are multiple possible errors in the same method, I disagree with re-using / overloading on the same variable.  As the method is now, sure, it would work, but generally speaking as methods get more complex it is to easy to confuse interpretation of the error.